packages/server/src/sync/api.ts

```diff
@@ -9,53 +10,38 @@ import { makeSyncRequestSchema } from './schema'
 import { SyncService } from './service'
 
 export class SyncApi {
-  constructor(
-    private router: Router,
-    private auth: Auth,
-    private syncs: SyncService,
-    private clients: ClientService,
-    private channels: ChannelService
-  ) {}
-
-  async setup() {
-    const force = !!yn(process.env.SPINNED)
-
-    const requestSchema = makeSyncRequestSchema(this.channels.list())
-
-    this.router.post(
-      '/sync',
-      this.auth.public.auth(async (req, res) => {
-        const channelsWithoutEnabled: { [channelName: string]: any } = {}
-        for (const [channelName, channelConfig] of Object.entries<any>(req.body?.channels || {})) {
-          if (channelConfig.enabled === false) {
-            continue
-          }
-
-          channelsWithoutEnabled[channelName] = _.omit(channelConfig, ['enabled'])
-        }
-        const bodyWithoutEnabled = { ...(req.body || {}), channels: channelsWithoutEnabled }
-
-        const { error, value } = requestSchema.validate(bodyWithoutEnabled)
-        if (error) {
-          return res.status(400).send(error.message)
-        }
-
-        const sync = value as SyncRequest
-
-        // We forbid sync requests that act on existing clients (valid clientId) but don't provide the correct token
-        // A sync request will accept a incorrect client id (we assume the client was deleted a provide a new client id in response)
-        // A sync request will also accept no client id (we assume the caller wants a new client id and we send it as a response)
-        if (sync.id) {
-          const client = await this.clients.getById(sync.id)
-          if (client && (!sync.token || !(await this.clients.getByIdAndToken(sync.id, sync.token)))) {
-            return res.sendStatus(403)
-          }
-        }
-
-        const result = await this.syncs.sync(sync, { name: force })
-
-        res.send(result)
-      })
-    )
+  private force!: boolean
+  private schema!: Joi.ObjectSchema
+
+  constructor(private syncs: SyncService, private clients: ClientService, private channels: ChannelService) {}
+
+  setup(router: Router, auth: Auth) {
+    this.force = !!yn(process.env.SPINNED)
+    this.schema = makeSyncRequestSchema(this.channels.list())
```

Comment:
Why not put this in the constructor?

Reply:
It would work put in theory this function relies on data from another service which should only be used after that service was setup.